ppr-ui/src/composables/mhrRegistration/useMhrValidations.ts

```diff
@@ -13,6 +16,8 @@ export const useMhrValidations = (validationState: any) => {
 
   /** Is true when all flags are true in specified section. */
   const getStepValidation = (section: MhrSectVal): boolean => {
+    // If section is not part of validation model return true
+    if (!validationState[section]) return true
```

Comment:
I was considering this, at first I added logic individual components to not run getStepValidation() for the manufacturer registration types, for the steps that we are not validating, which was a bit of extra checks, but is more concrete.

For example I added in 4 different spots. `!isManufactuerRegistration.value &&  getStepValidation(...)`

Which I can foresee leading to a lot of checks in many places.

You are correct that It can lead to false positive if the model is not set up correctly or the wrong model is passed to the composable. However, I believe that proper testing and API validations might be able to capture that.

 I can look into adding logic to validation composable to verify the model passed to it is correct for the current flow.

However, if you prefer checking in individual components I can do that.

Reply:
Just for more context this one place where I was running into issues with it checking validations not on the model. 
`HomeOwnersReview`
![image](https://github.com/bcgov/ppr/assets/77707952/18869859-1482-4507-b6ab-781a6aaaad95)

In the manufacturer flow we will have the HomeOwnersReview component with data fetched from database, but we don't have the add home owners step where we would be setting the validation.

Reply:
For separation of concerns and simplicity, can we just call a different step validation function (with its own validation model) when validating steps for Manf. Reg?

Reply:
The composable is agnostic of the model, so it can be reused for any 'stepper' style filing we create. 
Functions for a specific type of validation model would sort of defeat the purpose of the reusable composable no? 

I think less is more and although reuse of anything can introduce some measure of complexity, it should always be the goal if we can.

Reply:
Ok, in that case, can we pass a different validation model, that would have the right sections to validate? Have you tried that @orelbn?

Reply:
I am not clear about what right sections to validate means.

Reply:
For example, if we have only two Steps in the flow, we can create a validation model with only these two steps. That way, we do not need to reuse other validation model (that has more Steps) and return true for Steps that do not exist.